### DISCUSSION TITLE
How to only create flash file system if it is not already created, otherwise mount

### DISCUSSION THREAD

**Discussion by RichardB-RT** [micropython/micropython#16450] [STM32 / Pyboard] [answered]
In boot.py the FAT file system is created. The problem that arises is that every time boot.py is run, the flash file system is erased. How can this be prevented. Is there a way to determine that the file system is already created? When the fs is already created it should only be mounted. This is the code in boot.py.
```import os, pyb, machine
os.umount('/flash')
p1 = pyb.Flash(start=0, len=512*1024)
p2 = pyb.Flash(start=512*1024)
os.VfsFat.mkfs(p1)
os.VfsLfs2.mkfs(p2)
os.mount(p1, '/flash')
os.mount(p2, '/data')```

**Comment by robert-hh** [comment]
You can try to mount the file system first and only recreate it, if mountingb fails. A typical sequence from _boot.py of various ports is:
```
import vfs, pyb

bdev = pyb.Flash(start=0, len=512*1024)
vfs.umount('/flash')
try:
    vfs.mount(vfs.VfsFat(bdev), "/flash")
except:
    vfs.VfsFat.mkfs(bdev)
    vfs.mount(vfs.VfsFat(bdev), "/flash")
```

Note: For formatting code, either indent all code lines or enclose the code section into lines with three backticks ```

**Reply by RichardB-RT** [reply]
The last line gives an OSError: [Errno 1] EPERM on my board...

**Reply by robert-hh** [reply, marked as answer]
That's because the file system is already mounted. I added an umount like in your sample code. But the whole code is not needed.

**Comment by robert-hh** [comment]
There is no _boot.py in the STM32 port. But it uses the same logic in C code. So no, you do not have to create the on.board file system in your boot.py. The STM port will as well mount an SD card if present.

**Reply by RichardB-RT** [reply]
Thanks for helping. One question though. What name is the automatic file system mount reachable, because 
```
import os
os.listdir("/flash") 
```
does not seem to work?

**Comment by RichardB-RT** [comment]
```
import os
os.listdir('/')
```
does work and shows that flash partition is there
